src/kani-compiler/rustc_codegen_kani/src/codegen/function.rs

```diff
@@ -263,20 +313,64 @@ impl<'tcx> GotocCtx<'tcx> {
             mangled_name,
             original_file: loc.filename().unwrap(),
             original_line: loc.line().unwrap().to_string(),
+            unwind_value: None,
         };
 
-        self.proof_harnesses.push(harness);
+        harness
+    }
+
+    /// Unwind strings of the format 'unwind_x' and the mangled names are to be parsed for the value 'x'
+    fn handle_kanitool_unwind(&mut self, attr: &Attribute, harness: &mut HarnessMetadata) {
+        // Check if some unwind value doesnt already exist
+        if harness.unwind_value.is_none() {
+            // Get Attribute value and if it's not none, assign it to the metadata
+            if let Some(integer_value) = extract_integer_argument(attr) {
+                harness.unwind_value = Some(integer_value);
+            } else {
+                // Show an Error if there is no integer value assigned to the integer or there's too many values assigned to the annotation
+                self.tcx
+                    .sess
+                    .span_err(attr.span, "Exactly one Unwind Argument as Integer accepted");
+            }
+        } else {
+            // User warning for when there's more than one unwind attribute, in this case, only the first value will be
+            self.tcx.sess.span_err(attr.span, "Use only one Unwind Annotation per Harness");
+        }
     }
 }
 
 /// If the attribute is named `kanitool::name`, this extracts `name`
 fn kanitool_attr_name(attr: &ast::Attribute) -> Option<String> {
     match &attr.kind {
         ast::AttrKind::Normal(ast::AttrItem { path: ast::Path { segments, .. }, .. }, _)
```

Comment:
Since you are changing this function, can you please replace this logic by calling `attr.name_or_empty()`?